src/pages/index.js

```diff
@@ -1,75 +1,182 @@
+      cardSection.addItem(cardElement);
+    }
+  },
+  ".cards__list"
+);
+
+// Load user information and initial cards
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, initialCards]) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error("Error loading initial data:", err);
+  });
+
+// Handle profile form submission
+function handleProfileFormSubmit(formData) {
+  popupWithFormProfile.renderLoading(true); 
+  api.updateUserInfo(formData)
+    .then((updatedData) => {
+      userInfo.setUserInfo({
+        name: updatedData.name,
+        description: updatedData.about,
+        avatar: updatedData.avatar
+      });
+      popupWithFormProfile.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user profile:", err);
+    })
+    .finally(() => {
+      popupWithFormProfile.renderLoading(false); 
+    });
+}
+
+// Handle add card form submission
+function handleAddCardFormSubmit(formData) {
+  popupWithFormAddCard.renderLoading(true); 
+  api.addCard({
+    name: formData.title,
+    link: formData.url
+  })
+    .then((newCard) => {
+      const cardElement = createCard(newCard);
+      cardSection.addItem(cardElement);
+      popupWithFormAddCard.close();
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      popupWithFormAddCard.renderLoading(false); 
+    });
+}
+
+// Handle avatar form submission
+function handleAvatarFormSubmit(formData) {
+  popupWithFormAvatar.renderLoading(true); 
+  api.updateUserAvatar(formData.avatar)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        description: updatedUserData.about,
+        avatar: updatedUserData.avatar,
+      });
+      popupWithFormAvatar.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      popupWithFormAvatar.renderLoading(false);
+    });
+}
+
+// Handle card deletion
+function handleDeleteCard(cardId, cardElement) {
+  return api.deleteCard(cardId) 
+    .then(() => {
+      cardElement.remove();
+      popupWithConfirmation.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
+}
+
+// Create card
+function createCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    () => popupWithImage.open(data),
+    (cardId, cardElement) => popupWithConfirmation.open(cardId, cardElement),
+    api
+  );
+  return card.getView();
+}
+
+// Event listeners
+profileEditButton.addEventListener("click", () => {
+  const userInfoData = userInfo.getUserInfo();
+  popupWithFormProfile.setInputValues(userInfoData);
+  formValidators['profile-form'].resetValidation();
+  popupWithFormProfile.renderLoading(false); 
```

Comment:
`popupWithFormProfile.renderLoading(false);` is not needed on opening because you have it in `finally`